x-pack/platform/plugins/shared/streams/server/lib/tasks/task_client.ts

```diff
@@ -47,20 +49,27 @@ export class TaskClient<TaskType extends string> {
           space: '',
           stream: '',
           type: '',
+          task: {
+            params: {} as TParams,
+          },
         };
       }
 
       throw error;
     }
   }
 
+  // What if you want to cancel the current task and then re-schedule it?
```

Comment:
TODO: Add a cancel method, the route should also have a cancel flag so I know when to call this method (like when the user clicks "Stop" in the UI